x/bsonx/bsoncore/bsoncore_test.go

```diff
@@ -899,6 +900,60 @@ func TestBuild(t *testing.T) {
 	}
 }
 
+func TestNullBytes(t *testing.T) {
+	// Helper function to execute the provided callback and assert that it panics with the expected message. The
+	// createBSONFn callback should create a BSON document/array/value and return the stringified version.
+	assertBSONCreationPanics := func(t *testing.T, createBSONFn func() string, expected string) {
+		t.Helper()
+
+		var res string
+		defer func() {
+			got := recover()
+			assert.Equal(t, expected, got, "expected panic with error %v, got error %v with result %q", expected, got, res)
+		}()
+		res = createBSONFn()
+	}
+
+	t.Run("element keys", func(t *testing.T) {
+		createDocFn := func() string {
+			return NewDocumentBuilder().AppendString("a\x00", "foo").Build().String()
```

Comment:
To check, the panic occurs on the call to `AppendString`, right? The calls to `Build().String()` should never get reached. If so, I'd suggest removing them to tighten the test.

Similar question for the regex test cases.

Reply:
The additional calls were there so the tests would construct full BSON documents/values that could be printed out if the assertions in the helper function failed. I can see how this is not ideal, though, and these are unit tests that shouldn't require much debugging output if they fail, so I removed the extra calls.